pmd-java/src/test/java/net/sourceforge/pmd/lang/java/ast/JDKVersionTest.java

```diff
@@ -302,16 +302,21 @@ void jdk7PrivateMethodInnerClassInterface1() {
         List<ASTMethodDeclaration> methods = acu.findDescendantsOfType(ASTMethodDeclaration.class, true);
         assertEquals(3, methods.size());
         for (ASTMethodDeclaration method : methods) {
-            assertFalse(method.getEnclosingType().isInterface());
+            assertFalse(isEnclosingTypeInterface(method));
         }
     }
-
+    private boolean isEnclosingTypeInterface(ASTMethodDeclaration method) {
+        return method.getEnclosingType().isInterface();
+    }
     @Test
     void jdk7PrivateMethodInnerClassInterface2() {
         ParseException thrown = assertThrows(ParseException.class, () -> java7.parseResource("private_method_in_inner_class_interface2.java"));
-        assertTrue(thrown.getMessage().contains("line 19"));
+        assertTrue(isExceptionMessageContainsLine(thrown.getMessage(), "line 19"));
```

Comment:
Using the isExceptionMessageContainsLine method improves the readability and maintainability of the code and better expresses the meaning of the test assertions.

Because using thrown.getMessage().contains("line 19") to write assertion statements directly may be considered a hard-coded approach, which may lead to problems that are detrimental to code maintenance and reduced readability.

Reply:
Agreed, this logic has been written to very specific case and may not be scalable. However, since this is a test, it makes sense that the developers did not bother to put in too much effort as it is a development only code.